pandasai/helpers/output_types/__init__.py

```diff
@@ -0,0 +1,45 @@
+from typing import Union
+
+from ._output_types import (
+    NumberOutputType,
+    DataFrameOutputType,
+    PlotOutputType,
+    StringOutputType,
+    DefaultOutputType,
+)
+
+output_types_map = {
+    "number": NumberOutputType,
+    "dataframe": DataFrameOutputType,
+    "plot": PlotOutputType,
+    "string": StringOutputType,
+}
+
+
+def output_type_factory(
+    output_type,
+) -> Union[
+    NumberOutputType,
+    DataFrameOutputType,
+    PlotOutputType,
+    StringOutputType,
+    DefaultOutputType,
+]:
+    """
+    Factory function to get appropriate instance for output type.
+
+    Uses `output_types_map` to determine the output type class.
+
+    Args:
+        output_type (str): A name of the output type.
+
+    Returns:
+        (Union[
+            NumberOutputType,
+            DataFrameOutputType,
+            PlotOutputType,
+            StringOutputType,
+            DefaultOutputType
+        ]): An instance of the output type.
+    """
+    return output_types_map.get(output_type, DefaultOutputType)()
```

Comment:
@nautics889 I think this kind of validation might be useful to prevent unexpected behaviors!

Reply:
> I think this kind of validation might be useful to prevent unexpected behaviors!

This one was the most confusing for me.

Let me explain how i thought...

___

At first, it seemed appropriate to me to make an exception raising:
```python
def output_type_factory(output_type):
    if output_type is not None and output_type not in output_types_map:
        raise ValueError("Unknow 'output_type' has been passed. Possible are: ...")  # or custom exception
    return output_types_map.get(output_type, DefaultOutputType)()
```
In this way, user would be surely informed about pandasai can't give him a type which he/she expects.

However, the original requirements deny that:
> If output_type is not passed **or is not one of the specified values**, the instruction should remain as it is.

___

Well, then i try to log a warning message. Since we don't have a straight acces to pandasai logger, i've make the classic subpackage logging like this:
```python
import logging

...

logger = logging.getLogger(__name__)

...

def output_type_factory(output_type):
    if output_type is not None and output_type not in output_types_map:
        logger.warning("Unknow 'output_type' has been passed. Possible are: ...")
    return output_types_map.get(output_type, DefaultOutputType)()

```
Of course, this wouldn't give any to the stdout, since we [configure file handles with `.basicConfig`](https://github.com/gventuri/pandas-ai/blob/e06b3b977296aae5cd413041aa9217bb5c44b64d/pandasai/helpers/logger.py#L57) (i. e. only to a file). IDK, seems not too comprehensive for the end user. The logs are not appeared in `smart_df.logs` collection in this case ofc.
___
So, finally, i've add [an argument to pass for the logger object](https://github.com/gventuri/pandas-ai/pull/562/files#diff-db43e372e2106ddfbdea93579b194d767fa86e7c0b87971aa3a9894198a1210fR40); if passed some info about `output_type` is collected.
I wouldn't say it looks perfect for me, however we already have [something](https://github.com/gventuri/pandas-ai/blob/main/pandasai/smart_datalake/__init__.py#L95) [like](https://github.com/gventuri/pandas-ai/blob/main/pandasai/smart_datalake/__init__.py#L132) [this](https://github.com/gventuri/pandas-ai/blob/main/pandasai/smart_datalake/__init__.py#L401), so i guess it's OK.

___

Btw, how about to consider making [`Logger` object](https://github.com/gventuri/pandas-ai/blob/main/pandasai/helpers/logger.py#L33) a [singleton](https://en.wikipedia.org/wiki/Singleton_pattern), then we could import/instantiate/pass/whatever in any of submodules and get the same object each time? Of course, this is for further feature requests and "to be discussed".